### PR TITLE
Fix/add yamllint

### PR DESCRIPTION
**Pull request by wojtekzyla**
# Description

Added yamllint configuration file. Profiles and groups validation added to the code.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [x] Refactor/improvement

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

New integration tests

## Checklist

- [x] My commit message is [conventional](https://www.conventionalcommits.org/en/v1.0.0/#summary)
- [x] I have run pre-commit on all files before creating the PR
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: docs/bestpractices.md [hunk 1]
```diff
@@ -153,6 +153,17 @@ microk8s kubectl delete job/snmp-splunk-connect-for-snmp-inventory -n sc4snmp
 ```
 The upgrade command can be executed again. 
 
+### "The following profiles have invalid configuration" or "The following groups have invalid configuration" errors
+Example errors that might appear in case of wrong configuration:
```
Comment (ajasnosz) [inline docs/bestpractices.md:157]: Following errors are examples of wrong configuration:

File: docs/bestpractices.md [hunk 2]
```diff
@@ -153,6 +153,17 @@ microk8s kubectl delete job/snmp-splunk-connect-for-snmp-inventory -n sc4snmp
...
+```
+The following groups have invalid configuration and won't be used: ['group1']. Please check indentation and key words spelling inside these groups configuration.
```
Comment (ajasnosz) [inline docs/bestpractices.md:159]: inside these groups configuration -> inside mentioned groups configuration

File: docs/bestpractices.md [hunk 3]
```diff
@@ -153,6 +153,17 @@ microk8s kubectl delete job/snmp-splunk-connect-for-snmp-inventory -n sc4snmp
...
+```
+```
+The following profiles have invalid configuration and won't be used: ['standard_profile', 'walk_profile']. Please check indentation and key words spelling inside these profiles configuration.
```
Comment (ajasnosz) [inline docs/bestpractices.md:162]: similar here with wording

File: docs/bestpractices.md [hunk 4]
```diff
@@ -153,6 +153,17 @@ microk8s kubectl delete job/snmp-splunk-connect-for-snmp-inventory -n sc4snmp
...
+```
+These errors indicate, that the mentioned groups or profiles might have wrong indentation or some keywords were omitted or misspelled. Refer to [Configuring profiles](./configuration/configuring-profiles.md)
```
Comment (ajasnosz) [inline docs/bestpractices.md:164]: Errors above

File: docs/gettingstarted/sc4snmp-installation.md [hunk 5]
```diff
@@ -42,6 +42,22 @@ or download it directly from Helm using the command `microk8s helm3 show values
 
 It is recommended to start by completing the base template and gradually add additional configurations as needed.
 
+
+The `values.yaml` file is validated using `JSON schema` built into `helm chart` and additionally validated inside the code.
```
Comment (ajasnosz) [inline docs/gettingstarted/sc4snmp-installation.md:46]: and additionally validated inside the code. -> and inside the code.

File: docs/gettingstarted/sc4snmp-installation.md [hunk 6]
```diff
@@ -42,6 +42,22 @@ or download it directly from Helm using the command `microk8s helm3 show values
...
+If you want to make sure that your `values.yaml` follows formatting standards, you can use `yamllint`. In order to download
```
Comment (ajasnosz) [inline docs/gettingstarted/sc4snmp-installation.md:47]: To ensure that your ...

File: docs/gettingstarted/sc4snmp-installation.md [hunk 7]
```diff
@@ -42,6 +42,22 @@ or download it directly from Helm using the command `microk8s helm3 show values
...
+Then create `custom-config.yamllint` file and add the following configuration:
+```yaml
+extends: default
+
+rules:
+  line-length:
+    max: 80
+    level: warning
+
+```
```
Comment (ajasnosz) [inline docs/gettingstarted/sc4snmp-installation.md:58]: you can mention that it's also in our repo in examples or link it there

File: docs/gettingstarted/sc4snmp-installation.md [hunk 8]
```diff
@@ -42,6 +42,22 @@ or download it directly from Helm using the command `microk8s helm3 show values
...
+Next step is to run `yamllint -c <path to custom-config.yamllint> <path to values.yaml>`. Warnings can be ignored.
```
Comment (ajasnosz) [inline docs/gettingstarted/sc4snmp-installation.md:59]: Next run ... command.

File: integration_tests/test_poller_integration.py [hunk 9]
```diff
@@ -1268,6 +1268,87 @@ def test_wrong_profiles(self, request, setup_splunk):
         assert metric_count == 0
 
 
+@pytest.fixture(scope="class")
+def setup_misconfigured_profiles(request):
+    """
```
Comment (ajasnosz) [inline integration_tests/test_poller_integration.py:1273]: would it be possible to have similar test for groups?

**Review by ajasnosz — CHANGES_REQUESTED** [review]

File: docs/bestpractices.md (hunk 2 above)
Comment (wojtekzyla) [reply docs/bestpractices.md]: Done

File: docs/bestpractices.md (hunk 3 above)
Comment (wojtekzyla) [reply docs/bestpractices.md]: Done

File: docs/bestpractices.md (hunk 4 above)
Comment (wojtekzyla) [reply docs/bestpractices.md]: Done

File: docs/gettingstarted/sc4snmp-installation.md (hunk 5 above)
Comment (wojtekzyla) [reply docs/gettingstarted/sc4snmp-installation.md]: Done

File: docs/gettingstarted/sc4snmp-installation.md (hunk 6 above)
Comment (wojtekzyla) [reply docs/gettingstarted/sc4snmp-installation.md]: Done

File: docs/gettingstarted/sc4snmp-installation.md (hunk 7 above)
Comment (wojtekzyla) [reply docs/gettingstarted/sc4snmp-installation.md]: Done

File: docs/gettingstarted/sc4snmp-installation.md (hunk 8 above)
Comment (wojtekzyla) [reply docs/gettingstarted/sc4snmp-installation.md]: Done

File: docs/bestpractices.md [hunk 10]
```diff
@@ -153,6 +153,17 @@ microk8s kubectl delete job/snmp-splunk-connect-for-snmp-inventory -n sc4snmp
 ```
 The upgrade command can be executed again. 
 
+### "The following profiles have invalid configuration" or "The following groups have invalid configuration" errors
+Following errors are examples of wrong configuration:
+```
+The following groups have invalid configuration and won't be used: ['group1']. Please check indentation and key words spelling inside mentioned groups configuration.
```
Comment (ajasnosz) [inline docs/bestpractices.md:159]: keywords

File: docs/bestpractices.md [hunk 11]
```diff
@@ -153,6 +153,17 @@ microk8s kubectl delete job/snmp-splunk-connect-for-snmp-inventory -n sc4snmp
...
+```
+```
+The following profiles have invalid configuration and won't be used: ['standard_profile', 'walk_profile']. Please check indentation and key words spelling inside mentioned profiles configuration.
```
Comment (ajasnosz) [inline docs/bestpractices.md:162]: same here

**Review by ajasnosz — CHANGES_REQUESTED** [review]
Update changelog

File: docs/bestpractices.md (hunk 11 above)
Comment (wojtekzyla) [reply docs/bestpractices.md]: Done

File: docs/bestpractices.md (hunk 10 above)
Comment (wojtekzyla) [reply docs/bestpractices.md]: Done

**Review by ajasnosz — APPROVED** [review]

**Pull request merged**